lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+            self._arrange_interpolation_inputs(cube))
+        weights = WeightsUtilities.interpolate_to_find_weights(
+            source_points, target_points, source_weights,
+            fill_value=fill_value)
+        new_weights_cube = self._create_new_weights_cube(cube, weights)
+        return new_weights_cube
+
+    def process(self, cube):
+        """Calculation of linear weights based on an input configuration
+        dictionary. Weights are calculated for individually for each
+        point in self.config_coord_name. Weights are normalised across
+        the cube dimension specified by self.config_coord_name.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for setting up the interpolation and create the new weights
+                cube.
+
+        Returns:
+            new_weights_cube (iris.cube.Cube):
+                Cube containing the output from the interpolation. This
+                has been renamed using the self.weights_coord_name but
+                otherwise matches the input cube.
+        """
+        cube_slices = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over(self.config_coord_name):
+            new_weights_cube = (
+                self._interpolate_to_create_weights(cube_slice))
+            cube_slices.append(new_weights_cube)
+
+        # Normalise the weights.
+        new_weights_cube = cube_slices.merge_cube()
+        axis = new_weights_cube.coord_dims(self.config_coord_name)
+        new_weights_cube.data = (
+            WeightsUtilities.normalise_weights(
+                new_weights_cube.data, axis=axis))
+        return new_weights_cube
+
+
+class ChooseWeightsLinearFromCube(object):
+    """Calculate weights when providing an input cube containing a coordinate
+       over which the weights will be calculated e.g. forecast_period.
+       As an input cube is supplied, the weights can potentially vary
+       spatially.
+       The weights will be applied to a separate coordinate e.g. model_id, so
+       that the output will be a cube with different weights for each model_id
+       as the forecast period varies."""
+
+    def __init__(self, weighting_coord_name, config_coord_name):
+        """Set up for calculating linear weights from a cube of weights, where
+        for example, the weights vary along the forecast_period coordinate.
+
+        Args:
+            weighting_coord_name (str):
+                Name of the coordinate along which the weights will be
+                calculated. For example, if the intention is to provide
+                weights along the forecast_period coordinate within the
+                configuration dictionary, which can be linearly interpolated
+                to create all required weights, then this argument would be
+                "forecast_period".
+                This must be included within the configuration dictionary.
```

Comment:
Please remove references to "configuration dictionary" in this docstring - this plugin works with a weights cube.